packages/babel-plugin-transform-parameters/test/fixtures/regression/scope-gen-async/output.js

```diff
@@ -0,0 +1,23 @@
+function f() {
+  let a = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 1;
```

Comment:
Similar problem here, I don’t think a generator would throw until the first `next` call. I may be wrong though.

Reply:
Generators throw immediatly (`(function* (x = a.b) {})()` throws)